### DISCUSSION TITLE
FabGL is so marvelous...

### DISCUSSION THREAD

**Discussion by defdefred** [fdivitto/FabGL#166] [General]
I can't understand why it is not replacing bloated OS in all cash register or stock management console in every store...

Text mode is all we need for accounting management software...

What is the best text mode for the big 16/9 screen we all have nowdays?
Could ssh replace telnet ? https://github.com/ewpa/LibSSH-ESP32
With encrypted wifi + encrypted ssh => security is happy.

Thanks a lot fdivitto, you are a boss!

**Comment by niutech** [comment]
One problem is security - most connected devices require strong encryption like TLS 1.3, which requires more processing power. Another problem is auto-update in a secure manner, memory protection, process isolation, sandboxing, etc. Linux is a better choice in this area and it supports more programming languages.

**Reply by defdefred** [reply]
With double encrytion using SSH, TLS 1.2 is not such a problem.
ESP32 is able to do OTA update and the device is only used as ssh client terminal with real application running in a datacenter.
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/ota.html
